data/routers/action-router.js

```diff
@@ -8,108 +8,128 @@ router.get('/', async (req, res) => {
     const actions = await actionModel.get();
     res.status(200).json(actions);
   } catch (err) {
-    res
-      .status(500)
-      .json({ error: 'The actions information could not be retrieved.' });
+    res.status(500).json({
+      error: `There was an error while retrieving the actions. ${err}`
+    });
   }
 });
 
 router.get('/:id', async (req, res) => {
   try {
     const action = await actionModel.get(req.params.id);
-    res.status(200).json(action);
+    if (action) {
+      res.status(200).json(action);
+    } else {
+      res.status(404).json({
+        error: 'There is no action with the specified ID.'
+      });
+    }
   } catch (err) {
-    res.status(404).json({
-      error:
-        'There is no action with that ID, or there was an error retrieving the action information.'
+    res.status(500).json({
+      error: `There was an error while retrieving the action. ${err}`
     });
   }
 });
 
-router.post('/:id', async (req, res) => {
-  // First check if the project ID exists:
+router.post('/', async (req, res) => {
+  const { project_id, description, notes } = req.body;
   try {
-    project = await projectModel.get(req.params.id);
+    project = await projectModel.get(project_id);
+    if (!project) {
+      res.status(400).json({
```

Comment:
I wasn't sure what the standard is for this case. My thinking was that since the project_id was supplied by the user as a update to an existing action that this counted as bad request. The 404 is reserved (in this case) for trying to update an action whose id doesn't exist.